### ISSUE TITLE
Letter-Value plots

### ISSUE THREAD

**Issue by drewokane** [mwaskom/seaborn#338]
[Letter-Value plots paper](http://vita.had.co.nz/papers/letter-value-plot.pdf)

In the spirit of expanding the capabilities of Seaborn, I would like to propose adding letter-value plots. They are more robust than box plots for large data sets with long-tailed distributions and provide an alternative to violin plots which do not require the fitting of a kernel density function.

**Comment by drewokane** [comment]
To give you an update, since the upgrade of Seaborn to 0.6, I've refactored my original attempt at letter value plots to use the new `_CategoricalPlotter` class. It now works quite well. Here are a few examples using the Seaborn data sets (all the examples are taken from the [categorical plotting tutorial](http://stanford.edu/~mwaskom/software/seaborn/tutorial/categorical.html)):

1. The tips data set:

`letter_value.letter_value_plot(x="day", y="total_bill", data=tips);`

![image](https://cloud.githubusercontent.com/assets/9342082/8813759/adf533d0-2fba-11e5-905b-b3c0ec1bc089.png)

2. Using the `hue` parameter to nest plots:

`letter_value.letter_value_plot(x="day", y="total_bill", hue="time", data=tips);`

![image](https://cloud.githubusercontent.com/assets/9342082/8813793/fcd3313c-2fba-11e5-85d8-50053344c49a.png)

3. Sorting tips by size:

`letter_value.letter_value_plot(x="size", y="total_bill", data=tips.sort("size"));`

![image](https://cloud.githubusercontent.com/assets/9342082/8813816/29c9d312-2fbb-11e5-9c3d-010a0a89bc3b.png)

4. Flipping the axes:

`letter_value.letter_value_plot(x="total_bill", y="day", hue="time", data=tips);`

![image](https://cloud.githubusercontent.com/assets/9342082/8813908/ccb0da9e-2fbb-11e5-821e-2f0d970d2eda.png)

5. Plotting DataFrame objects:

`letter_value.letter_value_plot(data=iris, orient="h");`

![image](https://cloud.githubusercontent.com/assets/9342082/8813941/3980562c-2fbc-11e5-9306-f02c260b0174.png)

6. Using FacetGrid:

```
g = sns.PairGrid(tips,
                 x_vars=["smoker", "time", "sex"],
                 y_vars=["total_bill", "tip"],
                 aspect=.75, size=3.5)
g.map(letter_value.letter_value_plot, palette="pastel");
```

![image](https://cloud.githubusercontent.com/assets/9342082/8813954/5e21e504-2fbc-11e5-918f-65c06354e287.png)

As you can see, I've tried to keep the API consistent with the other plotting functions (boxplot, stripplot, etc), and preserve the functionality found in boxplot and violinplot.

I think letter value plots would be a wonderful addition to Seaborn for several reasons:

1. They preserve more information about the distribution of the data than boxplots do, and in most cases show the skewed nature of the data better.

2. Are a fast, non-parametric method for estimating the distribution of the data. I see letter value plots as being complimentary to violinplots, not replacements.

**Comment by mwaskom** [comment]
That looks great. I had mostly ignored this before (sorry) because I had been planning on doing a big refactor of the machinery for drawing categorical plots and that would have to be done before this could be reasonably implemented in seaborn.

I'm still a little ambivalent about including them, because this doesn't seem to be a very widely used approach (your pull request is the second thing that comes up on google when you search for them). While I see the arguments about their usefulness, the usefulness of a visualization is ultimately limited by familiarity too. So I'm wary about supporting (and answering questions) about it. But open to being convinced.

The other issue is licensing. Did you look at the ggplot code that implements Hadley's version before writing your own? While I'm not an expert on this topic, I think that would preclude your code from being included in seaborn, due to the GPL.

**Comment by drewokane** [comment]
In regards to the licensing issue, I did glance over Hadley's implementation before writing my own (as far as the implementation goes, I don't think I have the expertise to argue whether my implementation is substantially different that Hadley's...), I've checked and Hadley released the letter value plots for R under the MIT license. It seems that this would be compatible with the BSD license? Again, I'm speaking from inexperience here.

As far as familiarity goes, I agree that people generally use what they know (hence the wide use of boxplots). I do, however, think that if a visualization makes sense to people they will use it, even if they haven't used it before. Look at violin plots or boxplots. They *make* sense intuitively. You can interpret them directly as the density of the data. I may be shooting myself in the foot by saying all this but I believe letter value plots can be intuitive as well, in fact even more so than boxplots.

**Comment by mwaskom** [comment]
Sorry for the slow response, I've been traveling.

As long as the `lvplot` code is MIT I think that will be fine, license wise.

I still find the whole aspect of "letter values" to be confusing and hard to explain, but I guess your'e right that the plots make a fair amount of intuitive sense when you look at them, so I think they could be good, and we can just point people to the paper for a full explanation rather than having to explain it ourselves. If you want to put up another pull request, I can take a look at your code.
